File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/api/domain/requestModel/ResponseSchema.java
package com.mokatest.platform.demos.api.domain.requestModel;

import lombok.Data;

import java.util.List;

/**
 * 接口响应结构定义（对标 Apifox 数据模型绑定）
 *
 * 三种模式：
 *   NONE     — 不定义（默认，不校验）
 *   TEMPLATE — 绑定数据模板，overrides 可在引用处隐藏/覆盖字段
 *   INLINE   — 内联编辑 schema 规则树
 *
 * validateEnabled=true 时，执行后自动对响应体做结构校验（SchemaValidator），
 * 结果以 assertType=SCHEMA 的 AssertResult 混入 assertionResults。
 *
 * @author JingLong
 * @since 2026-08-06
 */
@Data
public class ResponseSchema {

    public enum Mode {
        NONE, TEMPLATE, INLINE
    }

    /**
     * 定义模式
     */
    private Mode mode = Mode.NONE;

    /**
     * 绑定的数据模板ID（mode=TEMPLATE 时有效）
     */
    private Integer templateId;

    /**
     * 内联 schema 规则树（mode=INLINE 时有效）
     */
    private MockFieldRule schema;

    /**
     * 引用处隐藏字段（mode=TEMPLATE 时有效）：字段名列表，校验/生成时从模板字段中剔除
     */
    private List<String> hiddenFields;

    /**
     * 引用处覆盖字段（mode=TEMPLATE 时有效）：与模板同名字段以本地规则为准
     */
    private List<MockFieldRule> overrideFields;

    /**
     * 执行时是否自动校验响应结构（默认 true）
     */
    private Boolean validateEnabled = true;
}

File: frontEnd/src/types/domain/api/requestModel/ResponseSchema.ts
import {MockFieldRule} from "@/types/domain/api/requestModel/MockFieldRule";

/**
 * 响应定义（对标 Apifox 数据模型）
 *
 * mode = NONE：不定义响应结构
 * mode = TEMPLATE：绑定数据模板，可隐藏/覆盖部分字段
 * mode = INLINE：内联定义响应结构
 */
export class ResponseSchema {
    /** 定义模式：NONE-不定义，TEMPLATE-绑定数据模板，INLINE-内联定义 */
    mode?: 'NONE' | 'TEMPLATE' | 'INLINE';

    /** 绑定的数据模板ID（mode=TEMPLATE 时有效） */
    templateId?: number;

    /** 内联响应结构（mode=INLINE 时有效，根节点 fieldType=OBJECT） */
    schema?: MockFieldRule;

    /** 隐藏字段名列表（mode=TEMPLATE 时有效）：从模板中剔除的字段 */
    hiddenFields?: string[];

    /** 覆盖字段列表（mode=TEMPLATE 时有效）：覆盖模板中同名字段 */
    overrideFields?: MockFieldRule[];

    /** 执行时是否校验响应结构 */
    validateEnabled?: boolean;

    constructor() {
        this.mode = 'NONE';
        this.templateId = undefined;
        this.schema = undefined;
        this.hiddenFields = [];
        this.overrideFields = [];
        this.validateEnabled = true;
    }
}
